src/hotspot/share/runtime/stackOverflow.hpp

```diff
@@ -95,18 +103,22 @@ class StackOverflow {
   //  |                                      |
   //  |  reserved zone                       |
   //  |                                      |
-  //  --  <-- stack_reserved_zone_base()    ---      ---
-  //                                                 /|\  shadow     <--  stack_overflow_limit() (somewhere in here)
-  //                                                  |   zone
-  //                                                 \|/  size
-  //  some untouched memory                          ---
+  //  --  <-- stack_reserved_zone_base()    ---   ---
+  //                                               ^
+  //                                               |    <--  stack_overflow_limit() [somewhere in here]
+  //                                               |  shadow
+  //                                               |   zone
+  //                                               |   size
+  //                                               v
+  //                                              ---   <--  shadow_zone_safe_limit()
   //
+  // (some usable, but not yet touched memory)
   //
   //  --
   //  |
   //  |  shadow zone
   //  |
-  //  --
+  //  --                                          ---   <--  shadow_zone_growth_watermark()
```

Comment:
Hi, should the `watermark` be somewhere below (regarding address) the last frame instead?

Reply:
Good point, fixed in new commit.